### DISCUSSION TITLE
Num classes in BBox head maskRCNN

### DISCUSSION THREAD

**Discussion by lauriebax** [open-mmlab/mmdetection#8884] [Q&A] [answered]
Hi,
When I tried to train my Mask RCNN on my custom data set I tried to set the class_weights for the CE-loss. It complained that it needed 8 weights, instead of the 7 classes I defined in my data set. Digging through the code I found this file [ConvFCBBoxHead](https://github.com/open-mmlab/mmdetection/blob/9d3e162459590eee4cfc891218dfbb5878378842/mmdet/models/roi_heads/bbox_heads/convfc_bbox_head.py#L86) where the head is made with `self.num_classes + 1`.

Can anyone explain to me why this is done and how I should interpret the additional class?

Thanks in advance

**Comment by zachcoleman** [comment, marked as answer]
It is a background class (i.e. your bbox head classifying it as BG/not-one-of-target-classes). See this code from base class:
https://github.com/open-mmlab/mmdetection/blob/9d3e162459590eee4cfc891218dfbb5878378842/mmdet/models/roi_heads/bbox_heads/bbox_head.py#L286-L287
